### DISCUSSION TITLE
SWG with vsfPump

### DISCUSSION THREAD

**Discussion by johnwillyn** [aqualinkd/AqualinkD#213] [General]
I completed my more permanent installation today, and decide to try a few of the config options that apply to my installation.

Specifically, I have a salt-water chlorine generator and a Pentair VS pump.

I enabled these two flags in /etc/aqualinkd.conf:

read_RS485_swg = yes
read_RS485_vsfPump = yes

I also set this:

force_SWG = yes

When I restarted things, my panel (perhaps the Aquapure controller) started beeping every 30 seconds or so, and flashing 0% in the panel LCD.

I know the docs say this is somewhat expected, but I was not expecting the beeping.

I am playing around with the different config settings now, but wondered if there was another setting or known combination of settings that could still report these settings, but not produce the audible alarm.

I am still (obviously) in the learning phase, so please forgive me if this has been covered before.

**Comment by sfeakes** [comment]
Those settings should not cause any difference in your panel. If you set them all to off do you see any difference? Ie beeping stops.

BTW none of them are needed they will simply allow aqualinkd to read information a lot quicker.

**Reply by johnwillyn** [reply]
If I leave them all out, the SWG does not show up in the UI at all.  Perhaps I just need to let it run for a bit so that it processes messages for SWG without these settings?

It appears that if I leave out this, but turn on SWG, the beeping does stop.

read_RS485_vsfPump = yes

I look forward to going deeper on some of these options and learning more.

JohnL

**Reply by sfeakes** [reply]
Deleted. For the moment

**Reply by sfeakes** [reply]
Just to confirm, the below setting makes your SWG beep when AqualinkD is started?
```read_RS485_vsfPump = yes```

On the two SWG settings.  The panel will only send SWG information when the pump is running. So AqualinkD has no idea you have SWG until the pump is running and SWG is enabled. So if you start AqualinkD with the pump off, you have to wait for the pump to cycle on before AqualinkD knows you have a SWG. It will then display SWG info. So yes, waiting would have eventually shown SWG in AqualinkD.
The below will let AqualinkD know you have a SWG and it will always create one on startup.  That’s all that setting does. 
```
force_SWG = yes
```


The Panel is super slow at displaying any SWG% or PPM messages, so the below setting allows AqualinkD to also read SWG messages on the RS 485 buss so you get instant updates of what the panel has asked the SWG to do, and what the SWG has told the panel it’s doing / what it’s values are.
```
read_RS485_SWG = yes
```

Hope that explains the settings.      
But I’m still very confused about reading Jandy messages makes your SWG beep.   If that is accurate, would you mind setting aqualinkd to serial debug, and posting the logs of when the SWG is beeping?    It’ll probably be a number of pages of messages.

**Reply by johnwillyn** [reply]
Sure, I would be happy to try to reproduce and run in serial debug.  

I was hoping that aqualinkd might be able to display more info about the salt cell than the Jandy app.  I have two of these systems; one at our home and one at a vacation rental home, and these salt cells have been a real pain.  Always throwing "codes", which are reported as "general fault".  If I can get info about the fault, I can direct our pool folks what to do (like clean the cell).

I will look at the serial debug doc now.

**Reply by johnwillyn** [reply]
As hard as I have tried, with all different configurations, I can't seem to duplicate the state with the alarms.

At one point, I did throw the salt cell into a fairly chaotic state, but that was after by right-click on the SWG icon a few times.  It was reporting that it needed salt, and the panel LCD was mostly gibberish.

I will keep trying tomorrow, and will only report back after I have a set of repeatable steps or configuration settings.

**Reply by sfeakes** [reply]
AqualinkD will report the exact error code the SWG provides, if you have `read_RS485_SWG = yes`.  BUT even though there is a "standard" not many manufactures follow it, and even worse post the wrong messages.  Even Jandy's SWG seem to be inconsistant depending on the firmware version. So in the display of AqualinkD you may still see "general fault" in many instances as I took out the exact message since it was confusing people. But the exact error number will be posted to an MQTT message `aqualinkd/SWG/fullstatus`. 
(0 on, 1 no flow, 2 low salt, 4 high salt, 8 clean cell, 9 turning off, 16 high current, 32 low volts, 64 low temp, 128 check PCB, 255 off)

If you want the full SWG status in AqualinkD web ui, then I could put it back with a config option.

**Reply by johnwillyn** [reply]
You were right, the settings above did not cause the beeping behavior I reported a week ago.  

Here are some more clues that I am following (because it started beeping and alternating between 50% and 0% on the SWG).

I was in my side yard and heard the beeping, so I checked the panel and saw the SWG LCD alternating as above.

I came in to connect to the rPi over VPN, but could not; it seemed to be offline.

I rebooted the rPi by unplugging it, waiting and plugging it back in.

The beeping and alternating SWG % setting continued.

I came back in to the house, and connected to the rPi, and stopped aqualinkd; the beeping stopped.  I checked the panel, it was normal again.

I then decided to restart aqualinkd, and it did not cause any beeping.

I am assuming my rPi lost power or something, and rebooted, so aqualinkd restarted with the rPi, and that caused the beeping and alternating settings to start.

Thinking about the day I set this up a week ago, I probably did restart the rPi after install, and had the AquaLink already running, and this is what started the beeping.

I am going to test this sequence again (and again) to see if I can diagnose this further.  I have also setup an mqtt logger to see what that can tell me, but I will also turn on logging to see what it captures (if I can reproduce).

**Comment by sfeakes** [comment]
I think you might have a RS485 bus problem.  Everything you connect to the bus takes away power, it kind-a sounds like after a period of time the SWG loosed connection to the control panel, and that's when it starts beeping.  I'd check the connections / terminations and cable on the RS485 bus.  A few of people have had similar issues, all but one have been solved with better connections / wire.

**Comment by johnwillyn** [comment]
Thanks for responding with this suggestion.  I will definitely check the wiring on all RS485 buss connections.  I did discover some loose wires when installing the USB interface cable, and re-arranged some wiring down to the bus extender (instead of doubling up), but it sounds like a need to do more in this area.

Thanks again.

**Reply by johnwillyn** [reply]
Thanks again for all of the feedback, but I think I have narrowed this issued down, and I have aqualinkd running without these issues at this point.

I believe the beeping and switching of SWG from 50% to 0% might have been caused by a "race condition" and the other configuration issues I was chasing were nothing more than "red herrings".

My Aqualink and the rPi4 are on the same circuit, and while I was tightening the RS485 contact wires, I was turning off power at the circuit panel.  When I turned power back on, I think aqualinkd might have started before the panel was ready, and it was causing the issues with the SWG settings.

If I stop aqualinkd and restart it, then it seems to run fine with my current config.  For now, I have disabled the read_RS485_vsfPump to eliminate that as a possible cause, but I will re-enable it and restart to see if it changes anything.

Has anyone else noticed any issues when the Aqualink and aqualinkd are restarted simultaneously (as in a power outage or something)?

I am now able to turn my attention to using aqualinkd as a "bridge" to report issues with the salt cell through another rPi that I use as a hub with my esp32 based pool filler.

**Comment by sfeakes** [comment]
@johnwillyn 

I have added some more information to the wiki on things to check, you may want to read over it.
https://github.com/sfeakes/AqualinkD/wiki#Common-Problems

**Reply by johnwillyn** [reply]
Thanks sfeakes!  This is very helpful, and I think the second item about the Pentair VSP is exactly the issue I was having.

I currently have the read_RS485_vsfPump disabled, and have been running all weekend without issue.

I will enable the vsfPump tomorrow to isolate that case.  

I want to leave the SWG read enabled to get the extended status messages; is that only available with read_RS485_swg = yes?

Thanks again.

**Reply by sfeakes** [reply]
yes, to get the extended SWG status `read_RS485_swg = yes` must be set.  The control panel hides all messages otherwise.
In the release I just posted v2.3.0f, you will also get the exact message in the webUI as well. (So you'll see no-flow / needs clean / salt to high etc) in the SWG icon.  And if your SWG is one of the ones where the error no# doesn't match the default, you can change it in the `web\config.js` file.
